File: python exercises/iterative-loop.py
# Task: rewrite this function so that it uses a loop rather than recursion

def joinElements(array, joinString): 
    result = array[0]
    for i in range(1, len(array)):  # start at 1 as [0] index has already been initialized in result
        result += joinString + array[i]
    return result

print(joinElements(['s','cr','t cod', ' :) :)'], 'e'))

File: exercises/iterative-loop.js
// Task: rewrite this function so that it uses a loop rather than recursion

function joinElements(array, joinString) {
    result = array[0];
    for (let i = 1; i < array.length; i++) { // start at 1 as [0] index has already been initialized in result.
        result += joinString + array[i];
    }
    return result;
}
  
console.log(joinElements(['s','cr','t cod', ' :) :)'], 'e'));